### PR TITLE
[addons] remove deprecated CAddonDatabase::GetAddon(const std::string…

### PR DESCRIPTION
**Pull request by howie-f**
…& addonID, ADDON::AddonPtr& addon);

/*! @deprecated: use CAddonMgr::FindInstallableById */
bool GetAddon(const std::string& addonID, ADDON::AddonPtr& addon);

## Description
replace deprecated `CAddonDatabase::GetAddon()` with `CAddonMgr::FindInstallableById()`.
should be a good idea to do this in alpha phase

## What is the effect on users?
n/a

## Types of change

- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

### CODE REVIEW

File: xbmc/addons/AddonInstaller.cpp [hunk 1]
```diff
@@ -116,12 +116,10 @@ void CAddonInstaller::GetInstallList(VECADDONS &addons) const
   }
   lock.unlock();
 
-  CAddonDatabase database;
-  database.Open();
   for (std::vector<std::string>::iterator it = addonIDs.begin(); it != addonIDs.end(); ++it)
   {
     AddonPtr addon;
-    if (database.GetAddon(*it, addon))
+    if (CServiceBroker::GetAddonMgr().FindInstallableById(*it, addon))
```
Comment (ksooo) [inline xbmc/addons/AddonInstaller.cpp:122]: you could save some cpu cycles if you obtain the value of `CServiceBroker::GetAddonMgr()` once outside the loop, not on every iteration.

File: xbmc/addons/AddonInstaller.cpp [hunk 2]
```diff
@@ -1147,10 +1134,15 @@ bool CAddonUnInstallJob::DoWork()
 
   AddonPtr addon;
   CAddonDatabase database;
+
   // try to get the addon object from the repository as the local one does not exist anymore
   // if that doesn't work fall back to the local one
-  if (!database.Open() || !database.GetAddon(m_addon->ID(), addon) || addon == NULL)
+  if (!database.Open() ||
```
[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/addons/AddonInstaller.cpp:1140]: it's the call to
https://github.com/xbmc/xbmc/blob/8446ed9d763bb0f821bee0964eadcd5d70ad4d2a/xbmc/addons/AddonInstaller.cpp#L1151

File: xbmc/addons/AddonInstaller.cpp [hunk 3]
```diff
@@ -395,12 +388,12 @@ bool CAddonInstaller::CheckDependencies(const AddonPtr &addon,
         (!haveInstalledAddon && !optional))
     {
       // we have it but our version isn't good enough, or we don't have it and we need it
-      if (!database.GetAddon(addonID, dep) || !dep->MeetsVersion(versionMin, version))
+      if (!CServiceBroker::GetAddonMgr().FindInstallableById(addonID, dep) ||
+          !dep->MeetsVersion(versionMin, version))
```
[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/addons/AddonInstaller.cpp:392]: done

File: xbmc/addons/AddonInstaller.cpp (hunk 1 above)
Comment (howie-f) [reply xbmc/addons/AddonInstaller.cpp:122]: done

File: xbmc/addons/Repository.cpp [hunk 4]
```diff
@@ -371,7 +371,8 @@ bool CRepositoryUpdateJob::DoWork()
     for (const auto& addon : addons)
     {
       AddonPtr oldAddon;
-      if (database.GetAddon(addon->ID(), oldAddon) && addon->Version() > oldAddon->Version())
+      if (CServiceBroker::GetAddonMgr().FindInstallableById(addon->ID(), oldAddon) &&
+          addon->Version() > oldAddon->Version())
```
[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/addons/Repository.cpp:375]: done

File: xbmc/addons/AddonInstaller.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/addons/AddonInstaller.cpp:1141]: Ah, I missed that call. I would move the db.open directly before the first call that needs the db, then.

[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/addons/AddonInstaller.cpp:1141]: On second thought, the failing database.Open() check leading to `abddon = m_addon` seems actually logically wrong. There is no obvious relation anymore. The new code encapsulates that addons are stored in database (which is good).

[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/addons/AddonInstaller.cpp:1141]: absolutely. i'd suggest to apply this on top?
```diff
diff --git a/xbmc/addons/AddonInstaller.cpp b/xbmc/addons/AddonInstaller.cpp
index d80e0842cd..abb5fdf2a1 100644
--- a/xbmc/addons/AddonInstaller.cpp
+++ b/xbmc/addons/AddonInstaller.cpp
@@ -1137,7 +1137,6 @@ bool CAddonUnInstallJob::DoWork()
   }
 
   AddonPtr addon;
-  CAddonDatabase database;
 
   // try to get the addon object from the repository as the local one does not exist anymore
   // if that doesn't work fall back to the local one
@@ -1149,12 +1148,20 @@ bool CAddonUnInstallJob::DoWork()
 
   auto eventLog = CServiceBroker::GetEventLog();
   if (eventLog)
-    eventLog->Add(EventPtr(new CAddonManagementEvent(addon, 24144)));
-
-  CServiceBroker::GetAddonMgr().OnPostUnInstall(m_addon->ID());
-  database.OnPostUnInstall(m_addon->ID());
+    eventLog->Add(EventPtr(new CAddonManagementEvent(addon, 24144))); // Add-on uninstalled
 
-  ADDON::OnPostUnInstall(m_addon);
+  CAddonDatabase database;
+  if (database.Open())
+  {
+    CServiceBroker::GetAddonMgr().OnPostUnInstall(m_addon->ID());
+    database.OnPostUnInstall(m_addon->ID());
+    ADDON::OnPostUnInstall(m_addon);
+  }
+  else
+  {
+    CLog::Log(LOGERROR, "{}: Unable to execute OnPostUnInstall. Can't open add-on database",
+              __func__);
+  }
 
   if (m_recurseOrphaned == RecurseOrphaned::CHOICE_YES)
   {
```

[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/addons/AddonInstaller.cpp]: Not sure. Why is a successfully opened database a precondition for the call to `CServiceBroker::GetAddonMgr().OnPostUnInstall(m_addon->ID());` and `ADDON::OnPostUnInstall(m_addon);` ? No db needed for these two calls.

[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/addons/AddonInstaller.cpp]: Not sure. Why is a successfully opened database a precondition for the call to `CServiceBroker::GetAddonMgr().OnPostUnInstall(m_addon->ID());` and `ADDON::OnPostUnInstall(m_addon);` ? No db needed for these two calls.

[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/addons/AddonInstaller.cpp]: hrm, a bit messy it is :)
i see that `CServiceBroker::GetAddonMgr().OnPostUnInstall()` does rely on an active db, but internally via AddonManager.

`ADDON::OnPostUnInstall(m_addon)` invokes the PostUnInstall-handler of the addon itself (if there is one).

how bout this?
```c
// this one uses m_database of CAddonMgr anyway...
CServiceBroker::GetAddonMgr().OnPostUnInstall(m_addon->ID());

CAddonDatabase database;
if (database.Open()) // safety net that doesn't help much
  database.OnPostUnInstall(m_addon->ID());

ADDON::OnPostUnInstall(m_addon); // no idea what the callback does, if there is one
```

[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/addons/AddonInstaller.cpp]: > // this one uses m_database of CAddonMgr anyway...

Never ever talk in source file A about implementation details in source file B !  I bet you forget to update this comment when changing the implementation no longer to rely on the db.

TBH, I also do not see much value in the other comments in the snippet.

[the remark this answers is not in the document]
Comment (howie-f) [reply xbmc/addons/AddonInstaller.cpp]: the comments were only for clarification here. i wouldnt write any comment. do you think the sequence is correct?

[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/addons/AddonInstaller.cpp]: Yes, and thanks for the clarification.

**Review by ksooo — APPROVED** [review]
LGTM.

File: xbmc/addons/AddonInstaller.cpp [hunk 5]
```diff
@@ -395,12 +389,12 @@ bool CAddonInstaller::CheckDependencies(const AddonPtr &addon,
         (!haveInstalledAddon && !optional))
     {
       // we have it but our version isn't good enough, or we don't have it and we need it
-      if (!database.GetAddon(addonID, dep) || !dep->MeetsVersion(versionMin, version))
+      if (!dep->MeetsVersion(versionMin, version) ||
```
Comment (howie-f) [inline xbmc/addons/AddonInstaller.cpp:392]: @ksooo also a potential crash candidate. dep could be nullptr (if !haveInstalledAddon && !optional).
will pr later if you can‘t get to it

**Review by AlwinEsch — COMMENTED** [review]
A bit late and already in :sweat_smile:, but thanks a lot about it and looks perfect!

**Pull request merged**